src/message/message-parser.js

```diff
@@ -10,16 +11,7 @@ const C = require('../constants/constants')
  *
  * @returns {Object} actions
 */
-const actions = (function getActions () {
-  const result = {}
-  let key
-
-  for (key in C.ACTIONS) {
-    result[C.ACTIONS[key]] = key
-  }
-
-  return result
-}())
+const actions = new Set(Object.keys(C.ACTIONS).map(key => C.ACTIONS[key]))
```

Comment:
We are trying to move away from having message parsing edge cases spread across the codebase with a slightly more guarded parser. Plus the fact they can go through permissions with an invalid action is also a bit odd.

In the C++ client we actually ensure that the topic exists, and then that the action is relevant to that topic. It makes the code alot simpler because you only have to worry about invalid context ( like subscribing to same event twice ).